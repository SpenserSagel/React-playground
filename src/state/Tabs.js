import React from 'react';

class Tabs extends React.Component{

    static defaultProps = {
        tabs : []
    };
    state = {
        currentTabIndex: 0
    };

    handleButtonClick(index){
        this.setState({currentTabIndex:index})
    }

    renderButtons(){
        const buttons = this.props.tabs.map((tab, index) =>{
            return <button key={index} onClick={() =>this.handleButtonClick(index)}>{tab.name}</button>
        });
        return buttons;
    }

    renderContent(){
        const currentTab = this.props.tabs[this.state.currentTabIndex];
        return(
            <div className='content'>
                {currentTab.content}
            </div>
        );
    }

    render(){
        return(
            <div>
                {this.renderButtons()}
                {!!this.props.tabs.length && this.renderContent()}
            </div>
        )
    }
}

export default Tabs;