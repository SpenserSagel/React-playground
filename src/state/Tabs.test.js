import React from 'react'
import ReactDOM from 'react-dom'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Tabs from './Tabs'

describe(`Tabs Component`, () => {
  const tabsProp = [
    {name:'tab1',
    content:'some random stufffff1'},
    {name:'tab2',
    content:'some random stufffff2'},
    {name:'tab3',
    content:'some random stufffff3'}]

  it('renders without errors', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Tabs />, div);
    ReactDOM.unmountComponentAtNode(div);
  })

  it('renders empty given no tabs', () => {
    const wrapper = shallow(<Tabs />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders the first tab by default', () => {
    const wrapper = shallow(<Tabs tabs={tabsProp} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('changes tabs', () =>{
      const wrapper = shallow(<Tabs tabs={tabsProp}/>);
      wrapper.find('button').at(1).simulate('click');
      expect(toJson(wrapper)).toMatchSnapshot();

  })
})