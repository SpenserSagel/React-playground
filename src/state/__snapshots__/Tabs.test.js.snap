// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs Component changes tabs 1`] = `
<div>
  <button
    key="0"
    onClick={[Function]}
  >
    tab1
  </button>
  <button
    key="1"
    onClick={[Function]}
  >
    tab2
  </button>
  <button
    key="2"
    onClick={[Function]}
  >
    tab3
  </button>
  <div
    className="content"
  >
    some random stufffff2
  </div>
</div>
`;

exports[`Tabs Component renders empty given no tabs 1`] = `<div />`;

exports[`Tabs Component renders the first tab by default 1`] = `
<div>
  <button
    key="0"
    onClick={[Function]}
  >
    tab1
  </button>
  <button
    key="1"
    onClick={[Function]}
  >
    tab2
  </button>
  <button
    key="2"
    onClick={[Function]}
  >
    tab3
  </button>
  <div
    className="content"
  >
    some random stufffff1
  </div>
</div>
`;
